From 262828a805463b169916dfb0d725d28dbf40f882 Mon Sep 17 00:00:00 2001
From: Cody Peter Mello <cody.mello@joyent.com>
Date: Fri, 31 Aug 2018 16:48:37 +0000
Subject: [PATCH] MANATEE-411 ManateePrimaryResolver should handle an empty
 cluster Reviewed by: Joshua M. Clulow <jmc@joyent.com> Approved by: Joshua M.
 Clulow <jmc@joyent.com>

---
 manatee.js | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/manatee.js b/manatee.js
index 4007c8e..8fe9380 100644
--- a/manatee.js
+++ b/manatee.js
@@ -616,6 +616,11 @@ ManateePrimaryResolver.prototype.state_running = function (S) {
             urls: urls
         }, 'manatee topology changed');
 
+        if (urls.length === 0) {
+            /* In a brand-new cluster, there will be nothing yet. */
+            return;
+        }
+
         var primary = mod_url.parse(urls[0]);
 
         assert.strictEqual(primary.protocol, 'tcp:');
-- 
2.21.0

